### PR TITLE
115 update with infer queue

### PR DESCRIPTION
**Pull request by OpenVINO-dev-contest**

### CODE REVIEW

File: notebooks/115-async-api/115-async-api.ipynb [hunk 1]
```diff
@@ -429,14 +429,125 @@
     "\n",
     "plt.show()"
    ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## AsyncInferQueue\n",
+    "Asynchronous mode pipelines can be supported with a wrapper class called AsyncInferQueue. This class automatically spawns the pool of InferRequest objects (also called “jobs”) and provides synchronization mechanisms to control the flow of the pipeline. It is a simpler way to manage the infer request queue in Asynchronous mode."
```
Comment (eaidova) [inline notebooks/115-async-api/115-async-api.ipynb:438]: I think it makes sense to add link on documentation
```suggestion
    "Asynchronous mode pipelines can be supported with a wrapper class called [AsyncInferQueue](https://docs.openvino.ai/latest/openvino_docs_OV_UG_Python_API_exclusives.html#asyncinferqueue). This class automatically spawns the pool of InferRequest objects (also called “jobs”) and provides synchronization mechanisms to control the flow of the pipeline. It is a simpler way to manage the infer request queue in Asynchronous mode."
```

**Review by eaidova — APPROVED** [review]

File: notebooks/115-async-api/115-async-api.ipynb [hunk 2]
```diff
@@ -429,14 +429,125 @@
...
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def callback(infer_request, info) -> None:\n",
+    "    \"\"\"\n",
+    "    Define the callback function for postprocessing\n",
+    "    \n",
+    "    :param: infer_request: the infer_request object\n",
+    "            info: a tuple includes oringal frame and start time\n",
```
Comment (adrianboguszewski) [inline notebooks/115-async-api/115-async-api.ipynb:460]:
```suggestion
    "            info: a tuple includes original frame and starts time\n",
```

File: notebooks/115-async-api/115-async-api.ipynb [hunk 3]
```diff
@@ -429,14 +429,125 @@
...
+    "    :retuns:\n",
```
Comment (adrianboguszewski) [inline notebooks/115-async-api/115-async-api.ipynb:461]:
```suggestion
    "    :returns:\n",
```

File: notebooks/115-async-api/115-async-api.ipynb [hunk 4]
```diff
@@ -429,14 +429,125 @@
...
+    "            infer_queue.start_async({input_layer_ir.any_name: resized_frame}, (frame, start_time))\n",
+    "    except KeyboardInterrupt:\n",
+    "        print(\"Interrupted\")\n",
+    "    # Any different error\n",
+    "    except RuntimeError as e:\n",
+    "        print(e)\n",
+    "    finally:\n",
+    "        infer_queue.wait_all()\n",
+    "        player.stop()\n",
+    "    return True"
```
Comment (adrianboguszewski) [inline notebooks/115-async-api/115-async-api.ipynb:525]:
```suggestion
```

File: notebooks/115-async-api/115-async-api.ipynb [hunk 5]
```diff
@@ -429,14 +429,125 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "frame_number = 0\n",
+    "total_time = 0\n",
+    "inferqueue(source=video_path, flip=False, fps=30, skip_first_frames=800)\n",
+    "print(f\"average throuput isn async mode: {inferqueue_fps:.2f} fps\")"
```
Comment (adrianboguszewski) [inline notebooks/115-async-api/115-async-api.ipynb:544]:
```suggestion
    "print(f\"average throughput in async mode with async infer queue: {inferqueue_fps:.2f} fps\")"
```

**Review by adrianboguszewski — CHANGES_REQUESTED** [review]

File: notebooks/115-async-api/115-async-api.ipynb [hunk 6]
```diff
@@ -429,14 +429,124 @@
...
+    "            # Start the inference request with async infer queue \n",
+    "            infer_queue.start_async({input_layer_ir.any_name: resized_frame}, (frame, start_time))\n",
+    "    except KeyboardInterrupt:\n",
+    "        print(\"Interrupted\")\n",
+    "    # Any different error\n",
+    "    except RuntimeError as e:\n",
+    "        print(e)\n",
+    "    finally:\n",
+    "        infer_queue.wait_all()\n",
+    "        player.stop()\n",
```
Comment (adrianboguszewski) [inline notebooks/115-async-api/115-async-api.ipynb:524]:
```suggestion
    "        player.stop()",
```

File: notebooks/115-async-api/115-async-api.ipynb [hunk 7]
```diff
@@ -429,14 +429,124 @@
...
+    "            # Start the inference request with async infer queue \n",
+    "            infer_queue.start_async({input_layer_ir.any_name: resized_frame}, (frame, start_time))\n",
+    "    except KeyboardInterrupt:\n",
+    "        print(\"Interrupted\")\n",
+    "    # Any different error\n",
+    "    except RuntimeError as e:\n",
+    "        print(e)\n",
+    "    finally:\n",
+    "        infer_queue.wait_all()\n",
+    "        player.stop()",
```
Comment (adrianboguszewski) [inline notebooks/115-async-api/115-async-api.ipynb:524]:
```suggestion
    "        player.stop()"
```

**Review by adrianboguszewski — APPROVED** [review]

**Pull request merged**